### PR TITLE
Support Linux 6.8

### PR DESCRIPTION
**Pull request by jason50123**

### CODE REVIEW

**Review by jserv — CHANGES_REQUESTED** [review]
Read https://cbea.ms/git-commit/ and make the git commit message informative.

File: file.c [hunk 1]
```diff
@@ -96,10 +96,17 @@ static int simplefs_readpage(struct file *file, struct page *page)
...
  */
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(6, 8, 0)
+static int simplefs_writepage(struct page *page, struct writeback_control *wbc) {
+    struct folio *folio = page_folio(page); 
+    return __block_write_full_folio(page->mapping->host, folio, simplefs_file_get_block, wbc);
+}
+#else
 static int simplefs_writepage(struct page *page, struct writeback_control *wbc)
-{
+{
```
Comment (jserv) [inline file.c]: Indent via `clang-format -i`.

**Review by jserv — CHANGES_REQUESTED** [review]
You need to discuss the API changes across recent Linux kernel versions and ensure that simplefs is updated accordingly. The git commit message should be detailed and informative, rather than merely using "refine."

File: inode.c [hunk 2]
```diff
@@ -63,10 +63,18 @@ struct inode *simplefs_iget(struct super_block *sb, unsigned long ino)
     inode->i_ctime.tv_nsec = 0;
 #endif
 
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(6, 7, 0)
+    inode->__i_atime.tv_sec = (time64_t) le32_to_cpu(cinode->i_atime);
```
Comment (jserv) [inline inode.c]: The accessors should be used instead of modifying the members directly.

Comment (jason50123) [reply inode.c]: I've used the accessors in the newest commit!

File: file.c [hunk 3]
```diff
@@ -183,7 +192,11 @@ static int simplefs_write_end(struct file *file,
     /* Update inode metadata */
     inode->i_blocks = inode->i_size / SIMPLEFS_BLOCK_SIZE + 2;
 
-#if LINUX_VERSION_CODE >= KERNEL_VERSION(6, 6, 0)
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(6, 7, 0)
+    cur_time = current_time(inode);
+    inode->__i_mtime = cur_time;
```
Comment (jserv) [inline file.c]: Can you avoid accessing `__i_mtime`?

File: inode.c [hunk 4]
```diff
@@ -161,7 +167,12 @@ static struct dentry *simplefs_lookup(struct inode *dir,
     brelse(bh);
 
     /* Update directory access time */
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(6, 7, 0)
+    dir->__i_atime = current_time(dir);
```
Comment (jserv) [inline inode.c]: Ditto. `__i_atime` is supposed to be manipulated indirectly.

**Review by jserv — CHANGES_REQUESTED** [review]
Rebase the latest `master` branch.

**Comment by jserv** [comment]
Thank @jason50123 for contributing!

**Pull request merged**